### DISCUSSION TITLE
ERROR WITH THE INTERACTION SCRIPT

### DISCUSSION THREAD

**Discussion by achyuta22** [foundry-rs/foundry#8939] [Help] [answered]
``` DEPLOYMENT SCRIPT```
// SPDX-License-Identifier:MIT
pragma solidity ^0.8.18;
import {Script} from "../lib/forge-std/src/Script.sol";
import {MyNFT} from "../src/MyNFT.sol";
contract DeployMyNFT is Script{
    function run() external returns(MyNFT) {
        vm.startBroadcast();
        MyNFT mynft=new MyNFT();
        vm.stopBroadcast();
        return mynft;
    }

}
```INTERACTION SCRIPT```
// SPDX-LICENSE-IDENTIFIER:MIT
pragma solidity ^0.8.18;

import {Script} from "../lib/forge-std/src/Script.sol";
import {MyNFT} from "../src/MyNFT.sol";
import {DevOpsTools} from "lib/foundry-devops/src/DevOpsTools.sol";

contract MintMyNFT is Script {
      string public constant PUG_URI =
        "ipfs://bafybeig37ioir76s7mg5oobetncojcm3c3hxasyd4rvid4jqhy4gkaheg4/?filename=0-PUG.json";
         function run() external{
            
        address contractAddress = DevOpsTools.get_most_recent_deployment("MyNFT", block.chainid);
        mintNFTOnContract(contractAddress);}

        function mintNFTOnContract(address contractAddress) public{
                    vm.startBroadcast();

            MyNFT myNFT =MyNFT(contractAddress);
            myNFT.mintNft(PUG_URI);
                    vm.stopBroadcast();

        }
        
}
```FOUNDRY.TOML```
[profile.default]
src = "src"
out = "out"
libs = ["lib"]
fs_permissions = [
    { access = "read", path = "./images/" },
    { access = "read", path = "./broadcast" },
]
ffi = true

# See more config options
# https://github.com/foundry-rs/foundry/blob/master/crates/config/README.md#all-options
remapping = ['@openzeppelin/contracts=lib/openzeppelin-contracts/contracts']

I have deployed the script using forge create commnad and it orinted deployer and deployed address adn hash too but while i used forge script for interactions.sol it is giving a error 
 Compiling...
No files changed, compilation skipped
Traces:
  [1000338] → new MintMyNFT@0x5b73C5498c1E3b4dbA84de0F1833c4a029d90519
    └─ ← [Return] 4886 bytes of code

  [145145] MintMyNFT::run()
    ├─ [0] VM::readDir("./broadcast", 3) [staticcall]
    │   └─ ← [Return] [("", "C:/Users/Bharat Achyuta/OneDrive/Desktop/web3 projects/NFT/broadcast\\DeployMyNFT.s.sol", 1, true, false), ("", "C:/Users/Bharat Achyuta/OneDrive/Desktop/web3 projects/NFT/broadcast\\DeployMyNFT.s.sol\\11155111", 2, true, false), ("", "C:/Users/Bharat Achyuta/OneDrive/Desktop/web3 projects/NFT/broadcast\\DeployMyNFT.s.sol\\11155111\\dry-run", 3, true, false)]
    ├─ [0] VM::toString(11155111 [1.115e7]) [staticcall]
    │   └─ ← [Return] "11155111"
    ├─ [0] VM::toString(11155111 [1.115e7]) [staticcall]
    │   └─ ← [Return] "11155111"
    ├─ [0] VM::toString(11155111 [1.115e7]) [staticcall]
    │   └─ ← [Return] "11155111"
    └─ ← [Revert] revert: No deployment artifacts were found for specified chain

Error:
script failed: revert: No deployment artifacts were found for specified chain

**Comment by zerosnacks** [comment, marked as answer]
This error originates from `foundry-devops`: https://github.com/Cyfrin/foundry-devops/blob/47393d0a85ad9f6aa127ba2aed2bf9a7a7488bcf/src/DevOpsTools.sol#L56-L58
